flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/descriptors/ConnectTableDescriptor.java

```diff
@@ -136,14 +137,24 @@ public void createTemporaryTable(String path) {
 					" use registerTableSource/registerTableSink/registerTableSourceAndSink.");
 		}
 
+		Map<String, String> properties = new HashMap<>(toProperties());
+
+		// handle schema
 		Map<String, String> schemaProperties = schemaDescriptor.toProperties();
 		TableSchema tableSchema = getTableSchema(schemaProperties);
-
-		Map<String, String> properties = new HashMap<>(toProperties());
 		schemaProperties.keySet().forEach(properties::remove);
 
+		// handle partition keys
+		DescriptorProperties descriptor = new DescriptorProperties();
```

Comment:
I'm not an expert of descriptors, but this logic dealing with partition keys seems tedious to me. Should we `new DescriptorProperties` twice and add/delete properties back and forth?

Reply:
Another way is `DescriptorProperties` provided `removeXX` methods.

Reply:
Why do you need this code in the first place?

As far as I can tell there is no way to pass partitionKeys to the `ConnectTableDescriptor`. They will never be in the properties. We need a method for doing it. Once you add one you don't need to put the keys into properties. You may just pass them explicitly to the `CatalogTable`.

The reason why its different for the `TableSchema` is backwards compatibility. The schema was part of properties before and was passed as part of `SchemaDescriptor`. This is why we had to extract and remove it from the properties.

Moreover I am not sure if it makes sense to have the `CatalogTableBuilder`. It duplicates the functionality of `ConnectTableDescriptor` and actually was never part of the public API. This is a different case though.

Reply:
@dawidwys You are right. `ConnectTableDescriptor` should have `withPartitionKeys` just like `withSchema`.

About `CatalogTableBuilder`, I think it is just for build `CatalogTable`. User just want to build `CatalogTable` and register it to catalog. Unlike `ConnectTableDescriptor`, it needs `Registration` too.